backend/medtagger/api/scans/business.py

```diff
@@ -94,23 +96,43 @@ def get_slices_for_scan(scan_id: ScanID, begin: int, count: int,
         yield _slice, image
 
 
-def add_label(scan_id: ScanID, selections: List[Dict], labeling_time: LabelingTime) -> Label:
+def add_label(scan_id: ScanID, elements: List[Dict], labeling_time: LabelingTime) -> Label:
     """Add label to given scan.
 
     :param scan_id: ID of a given scan
-    :param selections: List of JSONs describing selections for a single label
+    :param elements: List of JSONs describing elements for a single label
     :param labeling_time: time in seconds that user spent on labeling
     :return: Label object
     """
     user = get_current_user()
-    label = LabelsRepository.add_new_label(scan_id, user, labeling_time)
-    for selection in selections:
-        position = LabelPosition(x=selection['x'], y=selection['y'], slice_index=selection['slice_index'])
-        shape = LabelShape(width=selection['width'], height=selection['height'])
-        LabelsRepository.add_new_label_selection(label.id, position, shape)
+    try:
+        label = LabelsRepository.add_new_label(scan_id, user, labeling_time)
+    except IntegrityError:
+        raise NotFoundException('Could not find Scan for that id!')
+    for element in elements:
+        add_label_element(element, label.id)
     return label
 
 
+def add_label_element(element: Dict[str, Any], label_id: LabelID) -> None:
+    """Add new Label Element for given Label.
+
+    :param element: JSON describing single element
+    :param label_id: ID of a given Label that the element should be added to
+    """
+    tool = element['tool']
+    if tool == LabelTool.RECTANGLE.value:
+        position = LabelPosition(x=element['x'], y=element['y'], slice_index=element['slice_index'])
+        shape = LabelShape(width=element['width'], height=element['height'])
+        try:
+            label_tag = LabelTagRepository.get_label_tag_by_key(element['tag'])
+        except NoResultFound:
+            raise NotFoundException('Could not find any Label Tag for that key!')
+        LabelsRepository.add_new_rectangular_label_element(label_id, position, shape, label_tag)
+    else:
+        raise InvalidArgumentsException('Tool is not supported!')
```

Comment:
I think it will be worth to tell explicitly which tool is not supported.

**Example:** `"XYZ" tool is not supported!`